IO/GeoFileParser.cpp

```diff
@@ -733,37 +757,39 @@ Goal* GeoFileParser::parseWaitingAreaNode(TiXmlElement * e)
      Log->Write("INFO:\t  Goal subroom_id: %d", subroom_id);
 
      Goal* wa = new WaitingArea();
```

Comment:
Hi, I hope you don't mind my suggestion for improvement. As i saw, the variable `wa` isn't used anywhere except for the return. My suggestion would be to create a `WaitingArea*` directly. With this you get rid of the dynamic cast. The return will automatically cast the `WaitingArea*` towards `Goal*` later on.   
```suggestion
     WaitingArea* waitingArea = new WaitingArea();
```